### PR TITLE
fix #702: Inconsistent checks for Username

### LINKED ISSUE #702: Inconsistent check for Username during Signing Up
**Summary:**

Inconsistent check for Username during Signing Up

**Steps to reproduce:**

1) Go to Sign Up
2) Fill the details (username with spaces)
3) Submit details

**Expected behaviour:**

Instead, error message should be shown.

**Observed behaviour:**

If the form is submitted, new user will not be able login to the app (check for spaces in user name is present in login activity)

**Device and Android version:**

Xiaomi, Marshmallow

**Screenshots:**

issue_username_issue_2

issue_username_space_check

### PR DESCRIPTION
**Pull request by miPlodder**
Fixes #702 

**( trimmed Username in Login Section )**
pr_username_consistency_1

**( make check for size and spaces in Sign Up Section )**
pr_username_consistency_2

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by miPlodder** [comment]
@dilpreet96 @satyan @droidchef Please review this PR.

**Comment by therajanmaurya** [comment]
@miPlodder string.trim() doesn't remove the spaces between the string so please use `replace.(" ", "")` and make the changes.

**Comment by miPlodder** [comment]
@therajanmaurya I have used `trim()` to remove leading and trailing spaces in Username, but for checking whether the Username contains spaces in between, I have made a check using `contains(" ")` on the line,

https://github.com/miPlodder/mifos-mobile/blob/3a6124eeac9303f5af567f27510a8490b8cdbd24/app/src/main/java/org/mifos/mobilebanking/ui/fragments/RegistrationFragment.java#L126

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/LoginActivity.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ public void showPasswordError(String error) {
     @OnClick(R.id.btn_login)
     public void onLoginClicked() {
 
-        final String username = tilUsername.getEditText().getEditableText().toString();
+        final String username = tilUsername.getEditText().getEditableText().toString().trim();
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/LoginActivity.java]: @therajanmaurya Here, I have used string.trim() to remove leading and trailing spaces that can be added by user unintentionally.

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/RegistrationFragment.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ public void registerClicked() {
                 payload.setPassword(etPassword.getText().toString());
             }
             payload.setPassword(etPassword.getText().toString());
-            payload.setUsername(etUsername.getText().toString());
+            payload.setUsername(etUsername.getText().toString().trim());
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/RegistrationFragment.java]: @therajanmaurya This line will execute when there is no space in between, that's why I have used string.trim() here.

**Comment by therajanmaurya** [comment]
@miPlodder That's the issue you are removing the leading and trailing spaces. we don't want in between spaces too. So please use replace method to check that is there any space present in between the user entered username?

**Comment by miPlodder** [comment]
@therajanmaurya Please review again. I have made the changes and resolved the conflict.

**Pull request merged**